File: Python/228.Summary Ranges.py
#有序数组，返回它的摘要集，连续数字作为一个摘要
#For example, given [0,1,2,4,5,7], return ["0->2","4->5","7"].
class Solution(object):
    def summaryRanges(self, nums):
        if nums==[]:    return []
        result , start , l = [] , nums[0] , len(nums)
        for i in range(1,l):
            if nums[i]!=nums[i-1]+1:
                string = str(start) if nums[i-1]==start else str(start)+"->"+str(nums[i-1])
                result.append(string)
                start = nums[i]
        string = str(start) if nums[-1]==start else str(start)+"->"+str(nums[-1])
        result.append(string)
        return result

s = Solution()
print(s.summaryRanges([0,1,2,3]))

File: C++/Array/228.Summary Ranges.cpp
/*
有序数组，分段返回数字字符串
*/


#include <iostream>
#include <vector>
#include <math.h>
#include <cstring>
#include <sstream>
using namespace std;

class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
    	int length = nums.size();
		vector<string> result;
		if (length == 0)	return result;
		string temp = "";		
		int start = nums[0];
		for (int i = 0; i < length - 1; ++i)
		{
			if (nums[i] + 1 != nums[i + 1])
			{
				temp = to_string(start);
				temp += (nums[i] == start) ? "" : "->" + to_string(nums[i]);
				result.push_back(temp);
				start = nums[i + 1];
			}
		}
		if (nums[nums.size() - 1] - start == 0)
			result.push_back(to_string(start));
		else
		{
			result.push_back(to_string(start) + "->" + to_string(nums[length - 1]));
		}
		return result;
    }
};

int main()
{
	Solution s;
	vector<int> nums;
	vector<string> out;
	nums.push_back(-1);
	nums.push_back(1);
	nums.push_back(2);
	nums.push_back(4);
	nums.push_back(5);
	nums.push_back(7);
	nums.push_back(8);
	out = s.summaryRanges(nums);
	for(int i=0; i < out.size(); ++i)
	{
		cout<<out[i]<<","<<endl;
	}
}